src/runtime_src/hip/api/hip_event.cpp

```diff
@@ -7,50 +7,62 @@
 #include "hip/hip_runtime_api.h"
 
 #include "hip/core/event.h"
+#include "hip/core/stream.h"
 
 namespace xrt::core::hip {
 
-static hipEvent_t
+static command_handle
 hip_event_create()
 {
-  throw std::runtime_error("Not implemented");
+  return insert_in_map(command_cache, std::make_shared<event>(nullptr));;
 }
 
 static void
 hip_event_destroy(hipEvent_t event)
 {
   throw_invalid_value_if(!event, "event passed is nullptr");
-  throw std::runtime_error("Not implemented");
+  command_cache.remove(event);
 }
 
 static void
 hip_event_record(hipEvent_t event, hipStream_t stream)
 {
   throw_invalid_value_if(!event, "event passed is nullptr");
   throw_invalid_value_if(!stream, "stream passed is nullptr");
-  throw std::runtime_error("Not implemented");
+  //auto hip_stream = get_stream(stream);It is commented will be uncomment after stream PR is merged
+  auto hip_event_cmd = command_cache.get(event);
```

Comment:
We need to push the command to the stream and populate all current commands to event